e2e/tests/patient-maintenance.e2e-spec.ts

```diff
@@ -0,0 +1,122 @@
+        ExpectsUtil.checkWidgetPresentAndDisplayed(mainPage.getAllergyOptionsButton(), 'Allergy Options button');
+        ExpectsUtil.checkWidgetPresentAndDisplayed(mainPage.getAllergyGrid(), 'Allergy grid');
+
+        ExpectsUtil.checkNumber(mainPage.getAllergyGrid().getNumberOfRows(), 'Number of Allergies', 1);
+    }
+
+    function createPatientItem(patientData: string[]) {
+        patientMaintenancePage.getButtonAdd().click();
+        patientDetailPage.checkPresentAndDisplayed();
+        patientDetailPage.getNameInput().setText(patientData[0]);
+        patientDetailPage.getSurnameInput().setText(patientData[1]);
+        patientDetailPage.getEmailInput().setText(patientData[2]);
+        patientDetailPage.getAddressStreetInput().setText(patientData[3]);
+        patientDetailPage.getAddressCityInput().setText(patientData[4]);
+        patientDetailPage.getAddressZipInput().setText(patientData[5]);
+        patientDetailPage.getAddressCoordinatesInput().setText(patientData[6])
+        patientDetailPage.getButtonSubmit().click();
+        patientMaintenancePage.checkPresentAndDisplayed();
+        ExpectsUtil.checkNumber(patientMaintenancePage.getPatientsGrid().getNumberOfRows(), 'Number of Patients', 1);
+    }
+
+    function assignAllegryToPatient(allergyNum: number) {
+        patientDetailPage.getAddButton().click();
+        patientAllergyDetailPage.checkPresentAndDisplayed();
+        patientAllergyDetailPage.getAllergyCombobox().click();
+        patientAllergyDetailPage.getAllergyList().get(0).click();
+
+/*
+        patientAllergyDetailPage.getAssertedDate().setValue('01/01/2019');
+        patientAllergyDetailPage.getLastOccurrenceDate().setValue('02/02/2019');
+*/
+        patientAllergyDetailPage.getAllergyNotes().setText('aslhslkjhdsalkjah');
+        patientAllergyDetailPage.getSubmitButton().click();
+        ExpectsUtil.checkNumber(patientDetailPage.getAllergyGrid().getNumberOfRows(), 'Number of Allergies', 1);
+    }
+
+    it('Access to the Patient Management Dialog', () => {
+        patientMaintenancePage.showNewPageAndCheckTitleAndButtons(patientMaintenancePage.title, patientMaintenancePage.buttons);
+        const titles = ['', 'Name', 'Surname', 'Email'];
+        expect(patientMaintenancePage.getPatientsGrid().getGridHeader()).toEqual(titles);
+    });
+
+    it('Assign allergy to a patient', () => {
+        const optionMenuDelete = 1;
+
+        patientMaintenancePage.getButtonClose().click();
+        NavigationService.navigateToAllergyMaintenancePage(mainPage);
+        createAllergyItem(['Name1', 'Signs1', 'Symptoms1']);
+
+        NavigationService.navigateToPatientMaintenancePage(mainPage);
+        createPatientItem(['Patient1', 'Surname1', 'email@kk.com', 'Sample St', 'Khartum', '112234', '12.123456 32.15246']);
+        patientMaintenancePage.getPatientsGrid().clickOnCell(0, 'name');
+        patientDetailPage.checkPresentAndDisplayed();
+        patientDetailPage.getTabs().selectTab(1);
```

Comment:
As already commented, not sure if this is thread-safe.  I don't understand how the test is waiting for the selectTab() method to be completed before proceeding with the further check.

The following example demonstrates that the code is NOT waiting for the _asyncCall()_ method to be completed: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function
Add a console.log() at the end of this example (just after the _asyncCall()_) method to see this more clearly.

Reply:
Thanks, I will deeply check it.